src/frontend/screens/Settings/MapSettings/OfflineAreas.tsx

```diff
@@ -0,0 +1,179 @@
+import { BottomSheetModalMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
+import * as React from "react";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import { ScrollView, StyleSheet } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+
+import { RED } from "../../../lib/styles";
+import {
+  BottomSheetContent,
+  BottomSheetModal,
+  useBottomSheetModal,
+} from "../../../sharedComponents/BottomSheetModal";
+import Button from "../../../sharedComponents/Button";
+import HeaderTitle from "../../../sharedComponents/HeaderTitle";
+import { ErrorIcon } from "../../../sharedComponents/icons";
+import Loading from "../../../sharedComponents/Loading";
+import { OfflineAreaCard } from "../../../sharedComponents/OfflineAreaCard";
+
+const m = defineMessages({
+  title: {
+    id: "screens.Settings.MapSettings.OfflineArea",
+    defaultMessage: "Offline Areas",
+  },
+  removeMap: {
+    id: "screens.Settings.MapSettings.removeMap",
+    defaultMessage: "Remove Map",
+  },
+  cancel: {
+    id: "screens.Settings.MapSettings.cancel",
+    defaultMessage: "Cancel",
+  },
+  clearDiagrams: {
+    id: "screens.Settings.MapSettings.clearDiagrams",
+    defaultMessage: "Are you sure you want to Clear diagrams",
+  },
+  subtitle: {
+    id: "screens.Settings.MapSettings.subtitle",
+    defaultMessage:
+      "This map and offline areas attached to it will be deleted. This cannot be undone",
+  },
+});
+
+interface OfflineArea {
+  id: string;
+  title: string;
+  zoomLevel: number;
+  levelOfDetail: string;
+}
+
+export const OfflineAreas: NavigationStackScreenComponent = ({
+  navigation,
+}) => {
+  const bgMapId = React.useRef("");
+  const { formatMessage: t } = useIntl();
+
+  const { closeSheet, isOpen, openSheet, sheetRef } = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  const [listOfflineAreas, setListOfflineAreas] = React.useState<
+    OfflineArea[]
+  >();
+
+  const { getParam } = navigation;
+
+  React.useEffect(() => {
+    //Typesafety?
+    bgMapId.current = getParam("mapId", "");
+
+    //To Do Api call to get offline areas
+    function getAllOfflineAreas(mapId: string): OfflineArea[] {
+      return [
+        {
+          id: "1",
+          title: "Offline Area 1",
+          zoomLevel: 44,
+          levelOfDetail: "Small Roads",
+        },
+        {
+          id: "2",
+          title: "Offline Area 2",
+          zoomLevel: 44,
+          levelOfDetail: "Small Roads",
+        },
+      ];
+    }
+
+    setListOfflineAreas(getAllOfflineAreas(bgMapId.current));
+  }, []);
+
+  return (
+    <React.Fragment>
+      <ScrollView style={[styles.container]}>
+        {listOfflineAreas === undefined ? (
+          <Loading />
+        ) : (
```

Comment:
at a glance (could be wrong based on how this component is navigated to), but may need to represent three states here:

1. No offline areas
2. Loading offline areas
3. Showing offline areas

Reply:
No, there is always an offline area in the current flow (you cant download a background map without offline areas for the current MVP, but down the line we may need that). At the moment Sabella has not made the design for no offline area, but we could potentially add it to the copy deck